### PR TITLE
Feature: option to reverse slot cycle order

### PR DESCRIPTION
**Pull request by Carrascado**
When you press a slot key multiple times, the menu has to cycle the selection between all the available weapons in that slot.

- In this PR I introduce an option to reverse the order in which this cycling is done.
- The easiest mode to test it is with the blocks mode, but it works in all modes.
- Besides, the current order is considered now as "reversed", since I think the new order is more natural, but of course I leave to the reviewer's opinion whether to keep it this way. **In summary, top-to-bottom is "normal", and bottom-to-top is reverse.**
- I think this PR adds the option requested here: https://github.com/mmaulwurff/gearbox/issues/25, but not only for blocks, but for all selection configurations.

### CODE REVIEW

**Comment by mmaulwurff** [comment]
As far as I understand, this PR solves only input part of #25 while leaving display unchanged. This PR has its uses, of course, just I wouldn't close #25 right away after this is merged.

**Review by mmaulwurff — CHANGES_REQUESTED** [review]

File: zscript/gearbox/weapon_menu.zs [hunk 1]
```diff
@@ -73,13 +73,26 @@ class gb_WeaponMenu
   bool selectSlot(int slot)
   {
     uint nWeapons = mWeapons.size();
-    for (uint i = 1; i < nWeapons; ++i)
-    {
-      uint index = (mSelectedIndex + nWeapons - i) % nWeapons;
-      if (mSlots[index] == slot && isInInventory(index) && !isHidden(mWeapons[index].getClassName()))
+
+    if (mOptions.isSlotCycleOrderReversed()) {
```
[the remark this answers is not in the document]
Comment (Carrascado) [reply zscript/gearbox/weapon_menu.zs]: That's a nice idea! I have refactored it and made sure it works by testing it again.

**Review by mmaulwurff — COMMENTED** [review]
+

**Review by mmaulwurff — COMMENTED** [review]
+

**Pull request merged**